.husky/pre-commit

```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+STAGED=`git diff --cached --name-only --diff-filter=d`
+GENERATED=0
+
+if grep -q src/interfaces/ast/map-ast.ts <<< "$STAGED" && ! grep -q src/interfaces/ast/map-ast.schema.json <<< "$STAGED" ; then
+  yarn run --silent generate_map_schema
```

Comment:
I'm not sure this is robust enough. If I change the types, regenerate the schemas, stage, then change the types again this won't catch outdated but staged schemas.

I think this could rather check `mtime` on both files and trigger generation if the schemas are older than the types.